Have implemented the class `Battleship` with one extra task - `Print …

N1khto opened the pull request:
…field`

N1khto committed:
Have implemented the class `Battleship` with one extra task - `Print field`

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
-        pass
+        self.decks = [
+            Deck(row, column) for row in range(start[0], end[0] + 1)
+            for column in range(start[1], end[1] + 1)
```
```suggestion
            Deck(row, column) 
            for row in range(start[0], end[0] + 1)
            for column in range(start[1], end[1] + 1)
```

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list[tuple]) -> None:
```
```suggestion
    def __init__(self, ships: list[tuple[int, int]]) -> None:
```

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
+        if this_deck.is_alive:
+            this_deck.is_alive = False
+            if not any(deck.is_alive for deck in self.decks):
+                self.is_drowned = True
```
```suggestion
            self.is_drowned = not any(deck.is_alive for deck in self.decks)
```
Simplify this

N1khto committed:
Fixes...

Grayder0152 commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
         # And update the `is_drowned` value if it's needed
-        pass
+        this_deck = self.get_deck(row, column)
+        if this_deck.is_alive:
```
What if `this_deck` is None?

Grayder0152 commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
 
-    def fire(self, location: tuple):
+    def fire(self, location: tuple) -> str:
         # This function should check whether the location
         # is a key in the `self.field`
         # If it is, then it should check if this cell is the last alive
         # in the ship or not.
```
remove this or add to doc string

Grayder0152 commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list[tuple[int, int]]) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
```
remove this or add to doc string

Grayder0152 commented on app/main.py:
```diff
@@ -1,34 +1,75 @@
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
```
remove this or add to doc string

N1khto committed:
Fixes:
- deleted all unnecessary comments;
- added None-check in `fire` method of `Ship` class.

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,34 +1,64 @@
+    def fire(self, row: int, column: int) -> None:
+        this_deck = self.get_deck(row, column)
+        if this_deck:
+            if this_deck.is_alive:
```
if is not needed as for me
```suggestion
```

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,34 +1,64 @@
+            for column in range(10):
+                ship = self.field[(row, column)]
+                if ship is None:
+                    print("~", "\t", end="")
```
```suggestion
                    print("~", end="\t")
```
Can you try like this? Also the same in other places

N1khto committed:
Fixes:
- deleted redundant `if` in `fire` method of `Ship` class;
- optimized prints in `print_field` method of `Battleship` class.

ArsenPidhoretskyi reviewed (approved):
Well done!
